plugins/omega/coq_omega.ml

```diff
@@ -1452,17 +1452,15 @@ let normalize_equation sigma id flag theorem pos t t1 t2 (tactic,defs) =
   else
     (tactic,defs)
 
-let pf_nf gl c = Tacmach.New.pf_apply Tacred.simpl gl c
+let nf = Tacred.simpl
 
-let destructure_omega gl tac_def (id,c) =
-  let open Tacmach.New in
-  let sigma = project gl in
+let destructure_omega env sigma tac_def (id,c) =
   if String.equal (atompart_of_id id) "State" then
     tac_def
   else
     try match destructurate_prop sigma c with
       | Kapp(Eq,[typ;t1;t2])
-	when begin match destructurate_type sigma (pf_nf gl typ) with Kapp(Z,[]) -> true | _ -> false end ->
+	when begin match destructurate_type env sigma (nf env sigma typ) with Kapp(Z,[]) -> true | _ -> false end ->
```

Comment:
Why leaving a `nf` here (and hence a `Tacred.simpl`), if `destructurate_type` is now doing a proper conversion comparison ?